x/staking/src/keeper/validators_and_total_power.rs

```diff
@@ -0,0 +1,89 @@
+pub use super::*;
+
+impl<
+        SK: StoreKey,
+        PSK: ParamsSubspaceKey,
+        AK: AccountKeeper<SK>,
+        BK: BankKeeper<SK>,
+        KH: KeeperHooks<SK>,
+    > Keeper<SK, PSK, AK, BK, KH>
+{
+    /// Load the last total validator power.
+    pub fn get_last_total_power<DB: Database, CTX: QueryableContext<DB, SK>>(
+        &self,
+        ctx: &CTX,
+    ) -> Option<Uint256> {
+        let store = ctx.kv_store(&self.store_key);
+        store.get(&LAST_TOTAL_POWER_KEY).map(|bytes| {
+            Uint256::from_be_bytes(bytes.try_into().expect("Unexpected conversion error."))
```

Comment:
Same as above with use of expect. If this can only happen because there's either a bug in your code or the data is corrupt then we have this trait  `gears::store::database::ext::UnwrapCorrupt;` to captue that.

Reply:
I added detailed explanation for each `expect` and updated some handlers.  Please, check it. If you'll notice any unclear explanation, add a comment, please,